### PR TITLE
Support Linux kernel 6.3+

### PR DESCRIPTION
**Pull request by NishadSaraf**
### Problem Addressed by the Commit

This commit includes support for Linux 6.3+ kernels and resolves issues related to compiler warnings.

### Bug/Issue Resolution

No bug or issue was discovered that required fixing. 

### Solution Implemented and Rejected Alternatives

To address the problem, I used conditional preprocessor macros to replace deprecated kernel APIs for the latest kernels.

### Associated Risks

No risks are associated with the changes made in this commit.

### Testing Conducted and Additional Testing Requested

I have tested this pull request using Linux kernel versions 5.15 and 6.4-rc2, and confirmed that the xbutil validate tests pass.

### Documentation Impact

No impact on documentation.

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c [hunk 1]
```diff
@@ -1992,7 +1992,12 @@ static int __init xocl_init(void)
 {
 	int		ret, i = 0;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
```
Comment (keryell) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c:1995]: Curious. This PR is about 6.3+ but this changes make me think about 6.4+.

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c [hunk 2]
```diff
@@ -413,7 +413,11 @@ static void __xocl_subdev_destroy(xdev_handle_t xdev_hdl,
 		case XOCL_SUBDEV_STATE_ACTIVE:
 		case XOCL_SUBDEV_STATE_OFFLINE:
 			device_release_driver(&pldev->dev);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
```
Comment (keryell) [inline src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c:416]: Is this 5.4 so?

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c [hunk 3]
```diff
@@ -1685,7 +1689,11 @@ int xocl_subdev_create_vsec_devs(xdev_handle_t xdev)
 			    (subdev_info.priv_data))->flash_type,
 			    FLASH_TYPE_QSPIPS, strlen(FLASH_TYPE_QSPIPS));
 			/* default is FLASH_TYPE_SPI, thus pass through. */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0)
```
Comment (keryell) [inline src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c:1692]: Same question here.

**Comment by maxzhen** [comment]
ok to test

**Review by maxzhen — APPROVED** [review]

**Review by xdavidz — APPROVED** [review]

**Review by jvillarre — APPROVED** [review]

**Comment by keryell** [comment]
So, should the title be "Support Linux kernel 6.4+" instead?
And what about the test for version 5.4 in the code? Is this a typo or some old back-fixes?

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c (hunk 3 above)
Comment (xdavidz) [reply src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c:1692]: It depends on the very first version that linux kernel start supporting the new APIs, we want the exact version  as a split line.

**Comment by NishadSaraf** [comment]
> So, should the title be "Support Linux kernel 6.4+" instead? And what about the test for version 5.4 in the code? Is this a typo or some old back-fixes?

The [vm_flags change](https://github.com/Xilinx/XRT/pull/7588/commits/0454e25013f9e5e2a3fdbf01d0cff88b10a51b04#diff-a4b92ce32c3f7511ce92fc02ee03944b84fc10eb6d026313b51df33593945e33R318) needs to be fixed for 6.3 kernels. As for the 5.4 fixes, they are fix compiler warnings that have been tested locally.

**Review by keryell — APPROVED** [review]
Thanks.
I wanted to check it was not a typo.

**Pull request merged**